qt/python/mantidqt/mantidqt/widgets/sliceviewer/models/model.py

```diff
@@ -480,12 +480,17 @@ def get_proj_matrix(self):
         if not ws_type == WS_TYPE.MATRIX:
             proj_matrix = np.eye(3)  # needs to be 3x3 even if 2D ws as columns passed to recAngle when calc axes angles
             if ws_type == WS_TYPE.MDH:
-                ndims = ws.getNumDims()
-                i_qdims = [idim for idim in range(ndims) if ws.getDimension(idim).getMDFrame().isQ()]
-                irow_end = min(3, len(i_qdims))  # note for 2D the last col/row of proj_matrix is 0,0,1 - i.e. L
-                for icol, idim in enumerate(i_qdims):
-                    # copy first irow_end components of basis vec are always Q
-                    proj_matrix[0:irow_end, icol] = list(ws.getBasisVector(idim))[0:irow_end]
+                # for histo try to find axes from log
+                try:
+                    expt_info = ws.getExperimentInfo(0)
+                    proj_matrix = np.array(expt_info.run().get(PROJ_MATRIX_LOG_NAME).value, dtype=float).reshape(3, 3)
+                except:
+                    ndims = ws.getNumDims()
+                    i_qdims = [idim for idim in range(ndims) if ws.getDimension(idim).getMDFrame().isQ()]
+                    irow_end = min(3, len(i_qdims))  # note for 2D the last col/row of proj_matrix is 0,0,1 - i.e. L
+                    for icol, idim in enumerate(i_qdims):
+                        # copy first irow_end components of basis vec are always Q
+                        proj_matrix[0:irow_end, icol] = list(ws.getBasisVector(idim))[0:irow_end]
```

Comment:
If you comment out this liune from your script
`# ws.getExperimentInfo(0).run().addProperty('W_MATRIX', list(np.eye(3).flatten()), True)`
Then the `except` branch is called but we have  situation where the basis vectors have not been set so `proj_matrix = zeros(3,3)`. This screws up the non-ortho limit setting - which throws this error
```
Traceback (most recent call last):
  File "c:\mantid\qt\python\mantidqt\mantidqt\widgets\sliceviewer\views\dataview.py", line 423, in on_non_orthogonal_axes_toggle
    self.presenter.nonorthogonal_axes(state)
  File "c:\mantid\qt\python\mantidqt\mantidqt\widgets\sliceviewer\presenters\presenter.py", line 301, in nonorthogonal_axes
    self.show_all_data_clicked()
  File "c:\mantid\qt\python\mantidqt\mantidqt\widgets\sliceviewer\presenters\base_presenter.py", line 33, in show_all_data_clicked
    self.set_axes_limits(*limits)
  File "c:\mantid\qt\python\mantidqt\mantidqt\widgets\sliceviewer\presenters\base_presenter.py", line 48, in set_axes_limits
    self._data_view.set_axes_limits(xlim, ylim)
  File "c:\mantid\qt\python\mantidqt\mantidqt\widgets\sliceviewer\views\dataview.py", line 485, in set_axes_limits
    self.ax.set_xlim(xlim)
  File "C:\mantid\external\src\ThirdParty\lib\python3.8\Lib\site-packages\matplotlib\axes\_base.py", line 3241, in set_xlim
    left = self._validate_converted_limits(left, self.convert_xunits)
  File "C:\mantid\external\src\ThirdParty\lib\python3.8\Lib\site-packages\matplotlib\axes\_base.py", line 3158, in _validate_converted_limits
    raise ValueError("Axis limits cannot be NaN or Inf")
ValueError: Axis limits cannot be NaN or Inf
```
Would it be possible to check for invalid basis vectors and default to `proj_matrix = eye(3)` in that instance?

Reply:
Invalid `proj_matrix` could probably be tested by checking the determinant is non-zero (within some tolerance)?

Reply:
Made changes so that Basis Vectors are used first, then if the determinant is zero, try to get the info from the log. If that doesn't work, it reverts back to the identity matrix.

Reply:
Thanks - that works well for us as long as you're sure you're never going to have workspaces with basis vectors set for which you want to read the matrix in the log!